client/rest/src/main/java/org/elasticsearch/client/RestClientBuilder.java

```diff
@@ -53,7 +53,7 @@
     public static final int DEFAULT_MAX_CONN_PER_ROUTE = 10;
     public static final int DEFAULT_MAX_CONN_TOTAL = 30;
 
-    static final String VERSION;
+    public static final String VERSION;
```

Comment:
Was this intentional? Looks like this is only used for the user agent header now -- does the user agent header not also need to be the non-snapshot version?

Reply:
Yes, this is intentional, so that the new Java client can know the version of LLRC it's running with. It will need it to build its own telemetry header.

About snapshot versions, the user-agent contains the full version (i.e. with `SNAPSHOT`, `beta`, etc) whereas the client meta header used for telemetry expects a `p` to indicate a prerelease version but doesn't not care about what exact prerelease version is used.

Note that there are two bug fixes in this PR:
- use the `p` suffix for the telemetry header on non-final releases as described above
- fix the version string used in `build.gradle` for variable substitution in the `org/elasticsearch/client/version.properties` resource file: the expression that was used _always_ had a `-SNAPSHOT` suffix.